### PR TITLE
chore: update googleapis SHA circa 2024-04-22

### PR DESCRIPTION
**Pull request by dbolduc**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14040)

### CODE REVIEW

File: external/googleapis/protolists/discoveryengine.list [hunk 1]
```diff
@@ -11,6 +11,7 @@
 @com_google_googleapis//google/cloud/discoveryengine/v1:engine_service.proto
 @com_google_googleapis//google/cloud/discoveryengine/v1:import_config.proto
 @com_google_googleapis//google/cloud/discoveryengine/v1:purge_config.proto
+@com_google_googleapis//google/cloud/discoveryengine/v1:recommendation_service.proto
```
Comment (coryan) [inline external/googleapis/protolists/discoveryengine.list:14]: That looks like a new service. Do you mind adding a tracking bug if that is indeed the case?

File: google/cloud/storagecontrol/v2/.repo-metadata.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
     "issue_tracker": "https://issuetracker.google.com/issues?q=componentid:187243%20status=open",
     "language": "cpp",
     "library_type": "GAPIC_AUTO",
-    "name_pretty": "Cloud Storage API",
+    "name_pretty": "Storage Control API",
```
Comment (coryan) [inline google/cloud/storagecontrol/v2/.repo-metadata.json:9]: Sounds like we may want to change some stuff in the README file.  I will send a PR if needed.

File: google/cloud/secretmanager/v1/.repo-metadata.json [hunk 3]
```diff
@@ -3,11 +3,11 @@
     "api_shortname": "secretmanager",
     "client_documentation": "https://cloud.google.com/cpp/docs/reference/secretmanager/latest",
     "distribution_name": "google-cloud-cpp",
-    "issue_tracker": "https://issuetracker.google.com/savedsearches/5654226",
+    "issue_tracker": "https://issuetracker.google.com/issues?q=componentid:784854%20status=open",
     "language": "cpp",
     "library_type": "GAPIC_AUTO",
     "name_pretty": "Secret Manager API",
-    "product_documentation": "https://cloud.google.com/secret-manager/docs",
+    "product_documentation": "https://cloud.google.com/secret-manager/docs/overview",
```
Comment (coryan) [inline google/cloud/secretmanager/v1/.repo-metadata.json:10]: Do we need to update the README.md or main.dox files too?

**Review by coryan — APPROVED** [review]

File: google/cloud/storagecontrol/v2/.repo-metadata.json (hunk 2 above)
Comment (dbolduc) [reply google/cloud/storagecontrol/v2/.repo-metadata.json:9]: > I will send a PR if needed.

I see you sent a PR.

File: external/googleapis/protolists/discoveryengine.list (hunk 1 above)
Comment (dbolduc) [reply external/googleapis/protolists/discoveryengine.list:14]: That new service was the motivation for this PR. I will generate it.

**Pull request merged**